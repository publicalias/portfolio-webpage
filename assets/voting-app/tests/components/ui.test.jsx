"use strict";

//local imports

const UI = require("../../scripts/components/ui");

const { testWrapper } = require("../test-helpers");

//global imports

const { testMock } = require("test-helpers/meta-tests");
const { reactTests } = require("test-helpers/react-tests");

//node modules

const { MemoryRouter } = require("react-router-dom");

//setup

beforeAll(reactTests.setup);

//ui

describe("ui", () => {

  const { testMount, testShallow } = testWrapper(UI, MemoryRouter);

  it("should match snapshot", () => {

    const { wrapper } = testShallow();

    expect(wrapper).toMatchSnapshot();

  });

  it("should call metaGetUser on mount", () => {

    const { props, wrapper } = testMount();

    const { actions: { metaGetUser } } = props;

    wrapper.mount();

    testMock(metaGetUser, []);

    wrapper.unmount();

  });

});
