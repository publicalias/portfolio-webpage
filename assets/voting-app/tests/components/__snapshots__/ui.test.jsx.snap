// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui should match snapshot 1`] = `
<div
  className="c-ui"
>
  <NavBar
    actions={
      Object {
        "formAddOption": [MockFunction],
        "formCreatePoll": [MockFunction],
        "formDiscardPoll": [MockFunction],
        "formRemoveOption": [MockFunction],
        "formSetAddText": [MockFunction],
        "formSetTitleText": [MockFunction],
        "formToggleConfirm": [MockFunction],
        "formTogglePrivate": [MockFunction],
        "listOpenView": [MockFunction],
        "listSetIndex": [MockFunction],
        "listSetSearchText": [MockFunction],
        "listSetSort": [MockFunction],
        "listSubmitSearch": [MockFunction],
        "listToggleFlag": [MockFunction],
        "listToggleHide": [MockFunction],
        "menuOpenForm": [MockFunction],
        "menuSetFilter": [MockFunction],
        "menuToggleConfirm": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaGetPolls": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "viewAddOption": [MockFunction],
        "viewCastVote": [MockFunction],
        "viewDeletePoll": [MockFunction],
        "viewOpenList": [MockFunction],
        "viewRemoveOption": [MockFunction],
        "viewSetAddText": [MockFunction],
        "viewToggleConfirm": [MockFunction],
        "viewTogglePrivate": [MockFunction],
        "viewToggleSettings": [MockFunction],
      }
    }
    data={
      Object {
        "errors": Array [],
        "form": Object {
          "add": "",
          "confirm": false,
          "options": Array [],
          "private": false,
          "title": "",
        },
        "list": Object {
          "filter": "all",
          "index": 0,
          "search": "",
          "searched": "",
          "sort": "new",
        },
        "loading": false,
        "menu": Object {
          "confirm": false,
        },
        "polls": Array [],
        "user": Object {},
        "view": Object {
          "add": "",
          "confirm": false,
          "settings": false,
        },
      }
    }
    history={Object {}}
  />
  <Sidebar />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/list"
  />
  <Route
    component={[Function]}
    path="/form"
  />
  <Route
    component={[Function]}
    path="/view"
  />
</div>
`;
